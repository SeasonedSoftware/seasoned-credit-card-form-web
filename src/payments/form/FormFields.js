import React, { Fragment } from 'react'
import filter from 'lodash/filter'

import Field from './Field'
import fields from './fields'

export default props => {
  const commonProperties = ({
    touched,
    errors,
    handleChange,
    handleBlur,
    setFieldValue,
    onFocus,
  }) => ({
    touched,
    errors,
    handleChange,
    handleBlur,
    setFieldValue,
    onFocus,
  })
  return (
    <Fragment>
      {filter(fields(props)).map(element => {
        return (
          <Field
            key={element.name}
            type={element.type || 'text'}
            value={props.values[element.name]}
            {...element}
            {...commonProperties({
              ...props,
            })}
          />
        )
      })}
    </Fragment>
  )
}
