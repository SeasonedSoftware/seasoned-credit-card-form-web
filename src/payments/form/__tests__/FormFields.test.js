import React from 'react'
import renderer from 'react-test-renderer'

import FormFields from '../FormFields'

jest.mock('../Field', () => props => <div {...props}>Field - {props.name}</div>)
jest.mock('../fieldsHelper', () => ({
  labels: {
    'pt-BR': {},
    'en-US': {},
  },
  numberMask: value => value,
}))

it('renders correctly', () => {
  const props = {
    values: {
      name: '',
      number: '',
      cvc: '',
      expiry: '',
    },
    language: 'pt-BR',
    foo: 'bar',
  }
  const tree = renderer.create(<FormFields {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
