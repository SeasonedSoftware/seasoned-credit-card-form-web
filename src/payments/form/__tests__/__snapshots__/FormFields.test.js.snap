// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    autoFocus={true}
    foo="bar"
    fullWidth={true}
    name="name"
    value=""
  >
    Field - 
    name
  </div>,
  <div
    foo="bar"
    fullWidth={true}
    mask=""
    name="number"
    value=""
  >
    Field - 
    number
  </div>,
  <div
    foo="bar"
    mask="99/99"
    name="expiry"
    placeholder="MM/AA"
    value=""
  >
    Field - 
    expiry
  </div>,
  <div
    foo="bar"
    mask="9999"
    maskChar=""
    name="cvc"
    placeholder="123"
    value=""
  >
    Field - 
    cvc
  </div>,
]
`;
